File: sample-videochat-java/app/src/main/java/com/quickblox/sample/videochat/java/adapters/AudioCallAdapter.java
package com.quickblox.sample.videochat.java.adapters;

import android.content.Context;
import android.text.TextUtils;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.recyclerview.widget.RecyclerView;

import com.quickblox.sample.videochat.java.R;
import com.quickblox.users.model.QBUser;

import java.util.List;

public class AudioCallAdapter extends RecyclerView.Adapter<AudioCallAdapter.ViewHolder> {
    private List<ReconnectingUserModel> usersList;
    private final LayoutInflater inflater;

    public AudioCallAdapter(Context context, List<ReconnectingUserModel> usersList) {
        this.usersList = usersList;
        this.inflater = LayoutInflater.from(context);
    }

    public void updateList(List<ReconnectingUserModel> usersList) {
        this.usersList = usersList;
        notifyDataSetChanged();
    }

    public ReconnectingUserModel getItemByUserId(int userId) {
        for (ReconnectingUserModel item : usersList) {
            if (item.getUser().getId().equals(userId)) {
                return item;
            }
        }
        return null;
    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View v = inflater.inflate(R.layout.audio_call_item, parent, false);
        return new ViewHolder(v);
    }

    @Override
    public void onBindViewHolder(AudioCallAdapter.ViewHolder holder, int position) {
        QBUser user = usersList.get(position).getUser();
        String name;
        if (TextUtils.isEmpty(user.getFullName())) {
            name = user.getLogin();
        } else {
            name = user.getFullName();
        }
        holder.nameView.setText(name);

        if(!TextUtils.isEmpty(usersList.get(position).getReconnectingState())){
            holder.statusView.setText(usersList.get(position).getReconnectingState());
        }
    }

    @Override
    public int getItemCount() {
        return usersList.size();
    }

    public static class ViewHolder extends RecyclerView.ViewHolder {
        private final TextView nameView;
        private final TextView statusView;

        public ViewHolder(View itemView) {
            super(itemView);
            nameView = (TextView) itemView.findViewById(R.id.name);
            statusView = (TextView) itemView.findViewById(R.id.status);
        }

        public void setStatus(String status) {
            statusView.setText(status);
        }

        public void setName(String name) {
            nameView.setText(name);
        }
    }
}

File: sample-videochat-kotlin/app/src/main/java/com/quickblox/sample/videochat/kotlin/adapters/AudioCallAdapter.kt
package com.quickblox.sample.videochat.kotlin.adapters

import android.content.Context
import android.text.TextUtils
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.quickblox.sample.videochat.kotlin.R

class AudioCallAdapter(context: Context?, private var usersList: List<ReconnectingUserModel>) :
    RecyclerView.Adapter<AudioCallAdapter.ViewHolder>() {
    private val inflater: LayoutInflater

    init {
        inflater = LayoutInflater.from(context)
    }

    fun updateList(usersList: List<ReconnectingUserModel>) {
        this.usersList = usersList
        notifyDataSetChanged()
    }

    fun getItemByUserId(userId: Int): ReconnectingUserModel? {
        for (item in usersList) {
            if (item.getUser().id == userId) {
                return item
            }
        }
        return null
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view: View = inflater.inflate(R.layout.audio_call_item, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val user = usersList[position].getUser()
        val name: String
        name = if (TextUtils.isEmpty(user.fullName)) {
            user.login
        } else {
            user.fullName
        }
        holder.setName(name)
        if (!TextUtils.isEmpty(usersList[position].getReconnectingState())) {
            holder.setStatus(usersList[position].getReconnectingState())
        }
    }

    override fun getItemCount(): Int {
        return usersList.size
    }

    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        private val nameView: TextView
        private val statusView: TextView

        init {
            nameView = itemView.findViewById<View>(R.id.name) as TextView
            statusView = itemView.findViewById<View>(R.id.status) as TextView
        }

        fun setStatus(status: String?) {
            statusView.text = status
        }

        fun setName(name: String?) {
            nameView.text = name
        }
    }
}